components/tabs/src/sync/engine.rs

```diff
@@ -144,9 +148,10 @@ impl<'a> SyncEngine for TabsEngine<'a> {
             };
             remote_tabs.push(tab);
         }
-        self.storage.replace_remote_tabs(remote_tabs);
         let mut outgoing = OutgoingChangeset::new("tabs", inbound.timestamp);
-        if let Some(local_tabs) = self.storage.prepare_local_tabs_for_upload() {
+        let storage = self.store.storage.lock().unwrap();
```

Comment:
I don't think this matters much, but it might help avoid foot-guns in the future if we held the mutex for as short as possible - something like:

```
let local_tabs = {
  // keep the mutex for as short as possible
  let storage = self.store.storage.lock().unwrap();
  storage.replace_remote_tabs(remote_tabs);
  storage.prepare_local_tabs_for_upload()
};
if let Some(local_tabs) = local_tabs { ...
```

(in practice the statements below are going to be quick, but it seems worth being more explicit about mutex lifetimes where we can)